code/controllers/subsystem/SSair.dm

```diff
@@ -399,27 +486,87 @@ SUBSYSTEM_DEF(air)
 	GLOB.plmaster = new /obj/effect/overlay/turf/plasma
 	GLOB.slmaster = new /obj/effect/overlay/turf/sleeping_agent
 
-/datum/controller/subsystem/air/proc/throw_error_on_active_roundstart_turfs()
-	// Can't properly test lavaland due to Init order issues and EVERYTHING being surrounded by rocks, as such we just ignore any turfs on that level
-	var/list/active_turfs_we_care_about = list()
-	var/z_level_to_exclude = level_name_to_num(MINING)
-	for(var/turf/is_lavaland_turf in active_turfs)
-		if(is_lavaland_turf.z != z_level_to_exclude)
-			active_turfs_we_care_about += is_lavaland_turf
-	if(!length(active_turfs_we_care_about))
+/datum/controller/subsystem/air/proc/bind_turf(turf/T, list/milla_tile = null)
+	var/datum/gas_mixture/bound_to_turf/B = new()
+	T.bound_air = B
+	B.bound_turf = T
+	if(isnull(milla_tile))
+		milla_tile = new/list(MILLA_TILE_SIZE)
+		get_tile_atmos(T, milla_tile)
+	B.copy_from_milla(milla_tile)
+	B.dirty = FALSE
+	bound_mixtures += B
+
+
+/// Similar to addtimer, but triggers once MILLA enters synchronous mode.
+/datum/controller/subsystem/air/proc/synchronize(datum/milla_safe/CB)
+	// Any proc that wants MILLA to be synchronous should not sleep.
+	SHOULD_NOT_SLEEP(TRUE)
+
+	if(is_synchronous)
+		var/was_safe = SSair.in_milla_safe_code
+		SSair.in_milla_safe_code = TRUE
+		// This is one of two intended places to call this otherwise-unsafe proc.
+		CB.private_unsafe_invoke()
```

Comment:
Safer than either of those. It'll generate runtimes, because the code explicitly asserts on in_milla_safe_code, and runtimes again in MILLA if you try to write during asynchronous mode, but the worst actual outcome is that however you tried to change the air in the tile doesn't actually happen.